[IMP] Purchase: Refining 'Suggested quantities' page (#18927)
odoo/documentation: content/applications/inventory_and_mrp/purchase/advanced/suggest.rst

.. |RFQ| replace:: :abbr:`RFQ (request for quotation)`

The *Suggest* feature allows users to quickly and automatically calculate push-based replenishment
needs by recommending quantities to order based on historical demand.

Configuration
=============

First, ensure that the **Purchase**, **Inventory**, and **Sales** apps are :ref:`installed

> <general/install>`. Additionally, a least one delivery order must have been :ref:`validated

Reviewer: ```suggestion
<general/install>`. Additionally, at least one delivery order must have been :ref:`validated
```

feature is vendor-specific, so each product needs a matching vendor for accurate purchase quantity
and price calculations.

Finally, products to be used with the *Suggest* feature must have their *Product Type* set to
*Goods* and be :ref:`Tracked by quantity <inventory/product_management/manufacture>`. This ensures
the system can manage stock levels and calculate recommended replenishment quantities for tangible
items.

Suggest replenishment quantities
================================

To suggest quantities based on past sales, navigate to the :menuselection:`Purchase` app. Create a
:guilabel:`New` |RFQ| or select an existing one.

In the |RFQ|, set the :guilabel:`Vendor` field to the chosen supplier. In the *Products* tab, click
the :guilabel:`Catalog` button to view that vendor's items.

.. tip::
   By default, products listed in the product catalog are filtered by vendor. If products are
   missing, they may not be configured with the chosen vendor.

On the resulting page, toggle :guilabel:`Suggest` in the left sidebar to activate the feature. Two
fields appear underneath the toggle while it's active.

> - :guilabel:`Replenish for`: The duration (in days) from the current date that quantities are

Reviewer: Just a question about capitalization here. Later in the doc (line 92), you refer to the "Replenish For" days. Is there a capital F here, or does line 92 need to be edited to lowercase the "F"?

Author: Ah. Line 92 needs the tweak. Great catch.

  suggested for.
- :guilabel:`Based on`: The period used to determine demand: forecasted, last 7 days, 30 days, 3
  months, 12 months, or the same month or quarter the previous year. Additionally, the following
  *percentage* field defines the amount by which sales data is decreased or increased when
  suggesting replenishments (default 100%).

The total underneath these fields shows shows the order value to be added to the |RFQ|. Odoo
calculates this figure by multiplying the vendor's *Unit Price* by the suggested replenishment

> quantity.

Reviewer: ```suggestion
The total underneath these fields shows the order value to be added to the |RFQ|. Odoo calculates
this figure by multiplying the vendor's *Unit Price* by the suggested replenishment quantity.
```
You had a duplicate "shows" here.

Once the parameters are confirmed, click :guilabel:`Add All` to add all suggestions to the order.
Then, click :guilabel:`Back to Quotation` to confirm the final numbers on the |RFQ| and adjust the

> amounts of products to purchase (if needed.)

Reviewer: ```suggestion
amounts of products to purchase (if needed).
```

Misplaced period.

> .. _purchase/advanced/example-suggestion:

Reviewer: Not sure how far this would cascade to mean fixes in other docs, but you may want to update these anchors to use the proper structure: `app_name/file_name/anchor-name`.

Author: Turns out these links are outdated and nothing was looking for them. I've removed them completely.

.. example::

   A company needs to replenish orchids for the next 14 days, referencing the last 30 days of
   historical data at 100% of sales. Within those 30 days, the company delivered 20 units of orchids
   15 days ago and 20 units 1 day ago.

   .. math::

      Average~Daily~Demand = \frac{40}{30}  \approx 1.33 \text{ units/day}

   If the *Suggest* feature is used, the system recommends purchasing 19 orchids. Since the *Unit
   Price* for each orchid is $3, :math:`$3 \times 19 = $57`, which is the total amount displayed.

   .. math::

      Suggested~Quantity = 1.33 \times 14 \approx 18.67 \text{ (rounded to 19 units)}

   .. image:: suggest/result-14.png
      :alt: A screenshot showing the suggestion to purchase 19 units.

> .. _purchase/advanced/example-wh-suggestion:

Reviewer: This anchor is one you've created, so I would go ahead and update it to use the proper format: `app_name/file_name/anchor-name`.

Author: Oh, I copy-pasted this from elsewhere in the doc, I think. But I'll update all the anchors while I'm tweaking the page.

How Odoo calculates demand
==========================

To estimate demand, Odoo sums all :ref:`validated deliveries <inventory/delivery/one-step>`,
:ref:`components consumed in manufacturing orders <manufacturing/basic_setup/setup-components>`

> (MOs), or used to :doc:`resupply subcontractors

Reviewer: ```suggestion
(MOs), or components used to :doc:`resupply subcontractors
```

I had a hard time parsing this sentence. I *think* this is the correct word to insert here. If not, please add the correct word -- the comma-separated list doesn't make sense without something doing the "using."

<../../manufacturing/subcontracting/subcontracting_resupply>` coming from the warehouse specified in
the |RFQ| in the *Based on* period. The *average daily demand* is this sum of outgoing moves divided
by the number of days in the *Based on* period, multiplied by the *percentage*. The *estimated

> demand* is the *average daily demand* multiplied by *Replenish For* days.

Author: ```suggestion
demand* is the *average daily demand* multiplied by *Replenish for* days.
```

.. |RFQ| replace:: :abbr:`RFQ (request for quotation)`

Take the hard work out of calculating your replenishment needs with the *Suggest* feature. For
straightforward push-based replenishment strategies, the *Suggest* feature can recommend quantities
to order based on historical demand. Requests for quotations (RFQs) can be filled out in just a few

> clicks.

Reviewer 2: ```suggestion
The *Suggest* feature allows users to quickly and automatically calculate push-based replenishment needs by recommending quantities to order based on historical demand.
```

Reviewer 2: This sounds a little too sales-y/copywritten for documentation. Suggesting something like the above to keep the 'why' of the feature ('easy to use'/'automated') without veering too much into copy.

> Setting up the feature

Reviewer 2: ```suggestion
Configuration
```

Reviewer 2: Or 'Setup'. These seem to be the most consistent headings for anything setup-related. Also better for brevity.

======================

First, ensure that the **Purchase**, **Inventory**, and **Sales** apps are :ref:`installed
<general/install>`. Additionally, a least one delivery order must have been :ref:`validated
<inventory/delivery/one-step>` for each product to be considered for replenishment. This ensures
there is a past delivery record the system can use to calculate average daily demand.

Similarly, a vendor must be added to the :ref:`vendor pricelist
<purchase/manage_deals/vendor-pricelist>` with a purchase price for each product. The *Suggest*
feature is vendor-specific, so each product needs a matching vendor for accurate purchase quantity
and price calculations.

Finally, products to be used with the *Suggest* feature must have their *Product Type* set to
*Goods* and be :ref:`Tracked by quantity <inventory/product_management/manufacture>`. This ensures
the system can manage stock levels and calculate recommended replenishment quantities for tangible
items.

Suggest replenishment quantities
================================

To suggest quantities based on past sales, navigate to the :menuselection:`Purchase` app. Create a
:guilabel:`New` |RFQ| or select an existing one.

> In the |RFQ|, set the :guilabel:`Vendor` field to the chosen supplier. In the :guilabel:`Products`

Reviewer 2: ```suggestion
In the |RFQ|, set the :guilabel:`Vendor` field to the chosen supplier. In the *Products*
```

Reviewer 2: standard now is italics for tabs always.

tab, click the :guilabel:`Catalog` button to view that vendor's items.

.. tip::
   By default, products listed in the product catalog are filtered by vendor. If products are
   missing, they may not be configured with the chosen vendor.

> Inside the :guilabel:`Catalog`, toggle :guilabel:`Suggest` in the left sidebar to activate the

Reviewer 2: ```suggestion
On the resulting page, toggle :guilabel:`Suggest` in the left sidebar to activate the
```

feature. Two fields appear underneath the toggle while it's active.

- :guilabel:`Replenish for`: The period going forward that quantities are suggested for (measured in

>   days).

Reviewer 2: ```suggestion
- :guilabel:`Replenish for`: The duration (in days) from the current date that quantities are suggested for.
```

- :guilabel:`Based on`: The period used to define historical demand: last 7 days, 30 days, 3 months,

>   12 months, or the same month or quarter the previous year. Based on also has a *percentage* field,

Reviewer 2: :guilabel:`Forecasted` also shows up as an option for me.

  which is the amount by which sales data is decreased or increased when suggesting replenishments

>   (default 100%).

Reviewer 2: ```suggestion
  12 months, or the same month or quarter the previous year. Additionally, the following *percentage* field defines the amount by which sales data is decreased or increased when suggesting replenishments
  (default 100%).
```

Reviewer 2: Also, what kind of 'sales data'? Like the number of units sold, or revenue, etc.?

The total underneath these fields shows shows the order value to be added to the |RFQ|. Odoo
calculates this figure by multiplying the vendor's *Unit Price* by the suggested replenishment
quantity.

Once the parameters are confirmed, click :guilabel:`Add All` to add all suggestions to the order.

> hen click :guilabel:`Back to Quotation` to confirm the final numbers on the |RFQ| and adjust the

Reviewer 2: ```suggestion
Then, click :guilabel:`Back to Quotation` to confirm the final numbers on the |RFQ| and adjust the
```

amounts of products to purchase (if needed.)

.. _purchase/advanced/example-suggestion:

.. example::

   A company needs to replenish orchids for the next 14 days, referencing the last 30 days of
   historical data at 100% of sales. Within those 30 days, the company delivered 20 units of orchids
   15 days ago and 20 units 1 day ago.

   .. math::

      Average~Daily~Demand = \frac{40}{30}  \approx 1.33 \text{ units/day}

   If the *Suggest* feature is used, the system recommends purchasing 19 orchids. Since the *Unit
   Price* for each orchid is $3, :math:`$3 \times 19 = $57`, which is the total amount displayed.

   .. math::

      Suggested~Quantity = 1.33 \times 14 \approx 18.67 \text{ (rounded to 19 units)}

   .. image:: suggest/result-14.png
      :alt: A screenshot showing the suggestion to purchase 19 units.

.. _purchase/advanced/example-wh-suggestion:

> Behind the scenes

Reviewer 2: as much as i'd love to keep this heading, we should try to avoid idioms/colloquialisms for documentation as mentioned in the style guide for translation purposes.

> Behind the scenes

Reviewer 2: ```suggestion
How Odoo calculates demand
```

=================

Demand calculation

> ------------------

Reviewer 2: ```suggestion

```

To estimate demand, Odoo sums all :ref:`validated deliveries <inventory/delivery/one-step>`,
:ref:`components consumed in manufacturing orders <manufacturing/basic_setup/setup-components>`
(MOs), or used to :doc:`resupply subcontractors
<../../manufacturing/subcontracting/subcontracting_resupply>` coming from the warehouse specified in

> the RFQ in the *Based on* period. The *average daily demand* is this sum of outgoing moves divided

Reviewer 2: ```suggestion
the |RFQ| in the *Based on* period. The *average daily demand* is this sum of outgoing moves divided
```

by the number of days in the *Based on* period, multiplied by the *percentage*. The *estimated
demand* is the *average daily demand* multiplied by *Replenish For* days.

.. math::
   :class: overflow-scroll

   Average~Daily~Demand = \frac{Delivered~or~Consumed~Items}{Based~on~Days} \times Percentage
   \\ \\ Estimated~Demand = Average~Daily~Demand \times Replenish~for~Days

.. note::
   Only :guilabel:`Assigned` or :guilabel:`Confirmed` moves are taken into account in the *average
   daily demand* calculations. Sales quotations or manufacturing orders in :guilabel:`Draft` are not
   taken into account until validated.

.. tip::
   In a multi-warehouse setup, internal deliveries are also counted in demand estimation. In the
   case of a central warehouse dispatching products to individual stores, the *average daily demand*
   of the central warehouse includes internal transfers to individual stores.

To find the suggested replenishment quantity, Odoo deducts the *estimated demand* from the current
stock available quantity and all incoming shipments.

.. math::
   :class: overflow-scroll

   Recommended~Quantity = Estimated~Demand - (Available~Stock + Incoming~Stock)

> How to get the most out the *Suggest* feature

Reviewer 2: ```suggestion
Additional tips for usage
```

=============================================

Suggested replenishments are based on validated delivery orders, manufacturing orders, and other
inventory actions that consume quantities. For delivery orders, the *Effective Date* field is
considered the date the quantities were consumed.

Be sure to regularly review and update vendor pricelists to reflect the latest pricing and supplier
information to ensure correct suggestions.

Test sales projections based on seasonality. Reference prior months or quarters to capture seasonal
fluctuations and experiment with the *percentage* to project sales.

Always review suggestions critically. Although the *Suggest* feature provides a baseline
recommendation, always apply business judgment. Market changes, sales promotions, and upcoming

> events can all affect actual demand.

Reviewer 2: ```suggestion
recommendation, keep in mind that market changes, sales promotions, and upcoming
events can all affect actual demand.
```
